네이버는 파이썬 코드를 예제로 제공하는데 그 예제를 이용해서 간단하게 결과를 얻어 보겠다  
일단, API를 사용하기 위해 접근 url을 알아야 한다.  
각 검색 분야별 사용하는 URL은 다음과 같다. 

|종류|링크|
|:---:|:---:|
|블로그|https://openapi.naver.com/v1/search/blog.json|
|뉴스|https://openapi.naver.com/v1/search/news.json|
|책|https://openapi.naver.com/v1/search/books.json|
|백과사전|https://openapi.naver.com/v1/search/encyc.json|
|영화|https://openapi.naver.com/v1/search/movie/json|
|카페글|https://openapi.naver.com/v1/search/cafearticle.json|
|지식인|https://openapi.naver.com/v1/search/kin.json|
|웹문서|https://openapi.naver.com/v1/search/webkr.json|
|쇼핑|https://openapi.naver.com/v1/search/shop.json|

In [25]:
import urllib.request
client_id="fFs_uErHvCbOX3GdrzEj"
client_secret="BvThmXhJXe"

먼저 위 코드와 같이 naver api에 웹을 등록하면서 얻은 Client ID/secret 정보를 저장해둔다.  
여기서 사용된 
```
import urllib.request
```
urllib.request 모듈은 주어진 url을 이용해 웹에서 데이터를 얻어오는 역할을 하는 모듈이다.

In [26]:
encText = urllib.parse.quote("pinkwink")
url = "https://openapi.naver.com/v1/search/blog.json?query=" + encText
url

'https://openapi.naver.com/v1/search/blog.json?query=pinkwink'

네이버 API는 위 코드처럼 블로그 검색을 선택한 url에서 물음표(?) 후에 query를 붙여서 그 뒤에 나오는 단어를 검색한 결과를 얻게 된다.  
검색어로는 pinkwink를 사용하겠다.  
여기서
```
urllib.parse.quote("pinkwink")
```
라고 사용된 부분은 한글/영어 구분 없이 입력된 단어를 utf-8로 인코딩해서 url에서 사용할 수 있도록 해주는 기능이다.

In [27]:
request = urllib.request.Request(url)
request.add_header("X-Naver-Client-Id", client_id)
request.add_header("X-Naver-Client-Secret", client_secret)

이제 미리 만들어진 url을 urllib.request.Request 명령을 이용해서 응답을 얻어온다.  
이때, Client ID/secret 정보를 헤더(header)에 포함해서 보내면 인증과정도 지나가게 된다.

In [28]:
response = urllib.request.urlopen(request)
print(response.getcode())

200


In [29]:
type(response)

http.client.HTTPResponse

In [30]:
print(response.read().decode('utf-8'))

{
	"lastBuildDate":"Wed, 28 Jun 2023 18:12:13 +0900",
	"total":1250,
	"start":1,
	"display":10,
	"items":[
		{
			"title":"2022년 <b>PinkWink<\/b> 결산",
			"link":"https:\/\/pinkwink.kr\/1403",
			"description":"일상생활\/끄적거림 2022년 <b>PinkWink<\/b> 결산 <b>PinkWink<\/b> 2023. 1. 1. 06:00 2022년도 마지막 날입니다. 지금 저는 가족 여행으로 괌에 와 있는데요. 문득 올 한 해를 마무리하는 글을 작성하고 싶어지네요. 예전에는... ",
			"bloggername":"PinkWink",
			"bloggerlink":"https:\/\/pinkwink.kr\/",
			"postdate":"20230101"
		},
		{
			"title":"<b>PinkWink<\/b> darknet_ros, jetpack4.5.1에서 빌드 실패",
			"link":"https:\/\/blog.naver.com\/zeta0807\/222461538224",
			"description":"site\/BLDC-fbd4f8bd75424563b074d8adb3c858fa https:\/\/<b>pinkwink<\/b>.kr\/1342을 참고하면 Jetson Nano 에서... darknet_ros 다운로드 아래 명령어로 <b>PinkWink<\/b> darknet_ros를 내려받았다. 전체 로그는... ",
			"bloggername":"제타의 AIoT 나들이",
			"bloggerlink":"blog.naver.com\/zeta0807",
			"postdate":"20210808"
		},
		{
			"title":"<b>PinkWink<\/b> 블로그에서 온 선물..",
			"link":"https:\/\/blog.naver.com\/chandong83

response에 read()로 결과를 읽어서, 한글을 해석하기 위해 decode 옵션을 utf-8로 설정하면 pinkwink라는 단어를 네이버 블로그에서 검색한 결과가 나타난다.  
실습할 때는 결과가 다를 것인데, 실제로 검색하는 시점이 다르기 때문이다.  
또한, 전체 검색 결과가 2023년 6월 28일 18:12:13 기준으로 총 1250개, 시작이 1이고 표시(display)가 10이라고 되어 있다.  
검색 url을 만들 때, 이런 부분들이 더 보강될 수 있다.